interfaces/builtin/iio.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/snapcore/snapd/interfaces"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// The type for iio interface
+type IioInterface struct{}
+
+// Getter for the name of the iio interface
+func (iface *IioInterface) Name() string {
+	return "iio"
+}
+
+func (iface *IioInterface) String() string {
+	return iface.Name()
+}
+
+// Pattern to match allowed iio device nodes. It is gonna be used to check the
+// validity of the path attributes in case the udev is not used for
+// identification
+var iioControlDeviceNodePattern = regexp.MustCompile("^/dev/iio-device[0-9]")
```

Comment:
What is the actual matching string?  Tests are all using : instead of -. Please make sure tests are indeed verifying the logic for real, and breaking when they should. As it is, they certainly cannot be verifying the logic properly (search for "iio-device").